xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/resources/BaseSearchResult.java

```diff
@@ -212,84 +238,133 @@
 
             // Add ordering
             f.format(") order by %s", orderClause);
+            String queryString = f.toString();
 
-            String query = f.toString();
+            Locale userLocale = contextProvider.get().getLocale();
```

Comment:
You should use [``LocalizationContext#getCurrentLocale()``](https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-localization/xwiki-platform-localization-api/src/main/java/org/xwiki/localization/LocalizationContext.java#L38) instead, if that's the only reason you access the XWiki context.

Reply:
Thanks I was looking for something like this.